Runtime error due to POM - javax.inject [provided]

Hello,

I am configuring a SpringBoot application with Maven and QueryDSL for Hibernate.
Everything compiles and deploys correctly, the server starts, I starts a request, it opens the transaction...
But as soon as I do this :

`qd = new HibernateQueryFactory(sessionFactory.getCurrentSession());`

I get a runtime error :

```
java.lang.NoClassDefFoundError: javax/inject/Provider
    at com.asinter.agileqms.dao.MyDaoImpl.getHibernateQueryFactory(MyDaoImpl.java:22) ~[classes/:3.1.0-SNAPSHOT]
    at com.asinter.agileqms.dao.MyDaoImpl.getServiceType(MyDaoImpl.java:28) ~[classes/:3.1.0-SNAPSHOT]
    at com.asinter.agileqms.dao.MyDaoImpl$$FastClassBySpringCGLIB$$9f8a6888.invoke(<generated>) ~[classes/:3.1.0-SNAPSHOT]
    at org.springframework.cglib.proxy.MethodProxy.invoke(MethodProxy.java:204) ~[spring-core-4.3.3.RELEASE.jar:4.3.3.RELEASE]
    at org.springframework.aop.framework.CglibAopProxy$CglibMethodInvocation.invokeJoinpoint(CglibAopProxy.java:720) ~[spring-aop-4.3.3.RELEASE.jar:4.3.3.RELEASE]

Caused by: java.lang.ClassNotFoundException: javax.inject.Provider
    at org.apache.catalina.loader.WebappClassLoaderBase.loadClass(WebappClassLoaderBase.java:1285) ~[catalina.jar:8.5.5]
```

In the POM.XML, javax.inject is in the scope [provided] of querydsl-apt, as told in the documentation.
If I remove the scope, or if I add GUICE as a dependancy, it fixes the issue.

I'm using Tomcat 8.5

--- a/pom.xml
+++ b/pom.xml
@@ -136,6 +136,11 @@
         <version>${slf4j.version}</version>
         <scope>provided</scope>
       </dependency>
+      <dependency>
+        <groupId>javax.inject</groupId>
+        <artifactId>javax.inject</artifactId>
+        <version>1</version>
+      </dependency>
     </dependencies>
   </dependencyManagement>
 

--- a/querydsl-codegen/pom.xml
+++ b/querydsl-codegen/pom.xml
@@ -37,7 +37,6 @@
     <dependency>  
       <groupId>javax.inject</groupId>  
       <artifactId>javax.inject</artifactId>  
-      <version>1</version>  
     </dependency>
 
     <dependency>

--- a/querydsl-jpa/pom.xml
+++ b/querydsl-jpa/pom.xml
@@ -95,7 +95,6 @@
     <dependency>
       <groupId>javax.inject</groupId>
       <artifactId>javax.inject</artifactId>
-      <version>1</version>
     </dependency>
     
     <dependency>

--- a/querydsl-jpa/pom.xml
+++ b/querydsl-jpa/pom.xml
@@ -92,6 +92,11 @@
       <version>${project.version}</version>
       <scope>provided</scope>
     </dependency>
+    <dependency>
+      <groupId>javax.inject</groupId>
+      <artifactId>javax.inject</artifactId>
+      <version>1</version>
+    </dependency>
     
     <dependency>
       <groupId>com.querydsl</groupId>

